### ISSUE TITLE
`RecurrentVAE` uses the first RNN layer's hidden state and softmaxes twice

### ISSUE THREAD

**Issue by p3zo** [p3zo/rhythmic-relationships#22]
**1. Only the first RNN layer's hidden state is used.**

```python
# rhythmic_relationships/models/vae_recurrent.py:50-53
unused_output, hidden = self.input_to_hidden(x)
h = torch.cat((hidden[0], hidden[1]), dim=-1)
```

With `num_layers=2, bidirectional=True`, `hidden` has shape `(num_layers * num_directions, B, H)` = `(4, B, H)`. Indices 0 and 1 are **layer 0**'s forward/backward states; the final layer is at indices 2 and 3. So the second layer is trained but its output is discarded.

```
h.shape: (4, 2, 3)   # repo uses h[0], h[1]; last layer is h[2], h[3]
```

Should be `hidden[-2]`, `hidden[-1]`.

**2. Softmax followed by cross-entropy.**

`decode` returns `self.softmax(xoutput)` (`:61`), and `recurrent_part_vae_config.yml` sets `loss_fn: "cross-entropy"`. `nn.CrossEntropyLoss` applies `log_softmax` internally, so the distribution is softmaxed twice — flattening the gradient. Either return raw logits or switch to `nn.NLLLoss` on log-probabilities.

**3. `input_size=context_len`.**

```python
self.input_to_hidden = nn.RNN(input_size=context_len, ...)
```

The RNN's `input_size` is the per-timestep feature width, which should be `x_dim`, not the number of timesteps. The existing `TODO` at `:39` gestures at this confusion.

Also see #15 for the `sigma`-as-logvar problem this model shares.

**Comment by p3zo** [comment]
Items 1 and 2 are fixed in #45 — the encoder now reads `hidden[-2], hidden[-1]` (the last layer) and `decode` returns raw scores instead of softmaxing before a cross-entropy loss that softmaxes again.

Item 3 (`input_size=context_len`) is deliberately left open, because it isn't a one-line fix. The model is fed `PartDatasetSequential` output, which is **tokenized ids** — integers from the roll vocabulary — as continuous RNN features:

```python
x = xb.to(device).view(xb.shape[0] * xb.shape[1], xb.shape[2])   # (B*seq, context_len)
x_recon, mu, logvar = model(x)
```

With `batch_first=True`, a 2-D input is treated as *unbatched*, so the whole batch is consumed as one long sequence of `context_len`-wide feature vectors. Making this meaningful means deciding what a timestep and its features are — one tick with a pitch embedding? one tick of a raw roll? — which is a redesign of the model and its data path, not a bug fix.

Worth deciding whether `part_vae_recurrent` is still wanted at all before investing in it; if not, `archive/` is the honest home for it.
